agent/agentscripts/agentscripts.go

```diff
@@ -172,22 +178,31 @@ func (r *Runner) StartCron() {
 	}
 }
 
+type ExecuteOption int
+
+const (
+	ExecuteAllScripts ExecuteOption = iota
+	ExecuteStartScripts
+	ExecuteStopScripts
+	ExecuteCronScripts
+)
+
 // Execute runs a set of scripts according to a filter.
-func (r *Runner) Execute(ctx context.Context, filter func(script codersdk.WorkspaceAgentScript) bool) error {
-	if filter == nil {
-		// Execute em' all!
-		filter = func(script codersdk.WorkspaceAgentScript) bool {
-			return true
-		}
-	}
+func (r *Runner) Execute(ctx context.Context, option ExecuteOption) error {
 	var eg errgroup.Group
 	for _, script := range r.scripts {
-		if !filter(script) {
+		runScript := (option == ExecuteStartScripts && script.RunOnStart) ||
+			(option == ExecuteStopScripts && script.RunOnStop) ||
+			(option == ExecuteCronScripts && script.Cron != "") ||
+			option == ExecuteAllScripts
```

Comment:
Note: there doesn't seem to be test coverage in the `agentscripts` package for this, but I do see references in `agent/TestAgent_Lifecycle`. Maybe this is OK?

Reply:
I can break the logic here and see if any existing tests break, if not then I will add some tests.

Reply:
Breaking the logic here does break the `agent/TestAgent_Lifecycle` tests.

Reply:
Nice! We're covered then.